flang/include/flang/Optimizer/HLFIR/HLFIRDialect.h

```diff
@@ -82,6 +82,17 @@ inline bool isPolymorphicType(mlir::Type type) {
   return fir::isPolymorphicType(type);
 }
 
+/// Is this the FIR type of a Fortran procedure pointer?
+inline bool isFortranProcedurePointerType(mlir::Type type) {
```

Comment:
Why did you decide to give this a hlfir wrapper? Are you expecting this to diverge from fir::isBoxProcAddressType some time in the future?

Reply:
Not really, but HLFIR predicate names tends to use Fortran concepts, I find it easier to read/understand `isFortranProcedurePointerType` in lowering from the parse tree rather than to get that `fir::isBoxProcAddressType` implies that we are talking about procedure pointers because that is how they are implemented.